src/armTemplates/resources/functionApp.ts

```diff
@@ -39,6 +43,18 @@ export class FunctionAppResource implements ArmResourceTemplateGenerator {
         defaultValue: "",
         type: ArmParamType.String
       },
+      functionAppKind: {
+        defaultValue: "functionapp",
+        type: ArmParamType.String,
+      },
+      functionAppReserved: {
+        defaultValue: false,
```

Comment:
We should add some inline comments in the ARM template resources that describe the purpose of each of these properties.